src/snapshots/occlum/unionfs.rs

```diff
@@ -0,0 +1,188 @@
+// Copyright (c) 2022 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+
+// This unionfs file is used for occlum only
+
+use std::fs;
+use std::path::{Path, PathBuf};
+use std::sync::atomic::AtomicUsize;
+
+use anyhow::{anyhow, Result};
+use dircpy::CopyBuilder;
+use fs_extra;
+use fs_extra::dir;
+use nix::mount::MsFlags;
+
+use crate::snapshots::{MountPoint, Snapshotter};
+
+#[derive(Debug)]
+pub struct Unionfs {
+    pub data_dir: PathBuf,
+    pub index: AtomicUsize,
+}
+
+fn clear_path(mount_path: &Path) -> Result<()> {
+    let mut from_paths = Vec::new();
+    let paths = fs::read_dir(
+        mount_path
+            .to_str()
+            .ok_or(anyhow!("mount_path does not exist"))?,
+    )?;
+    for path in paths {
+        from_paths.push(path?.path());
+    }
+    fs_extra::remove_items(&from_paths)?;
+
+    Ok(())
+}
+
+fn create_dir(create_path: &PathBuf) -> Result<()> {
+    if !create_path.exists() {
+        fs::create_dir_all(create_path.as_path())?;
+    }
+
+    Ok(())
+}
+
+fn create_environment(mount_path: &Path) -> Result<()> {
+    let mut from_paths = Vec::new();
+    let mut copy_options = dir::CopyOptions::new();
+    copy_options.overwrite = true;
+
+    #[cfg(target_arch = "x86_64")]
```

Comment:
ooi, do we need something like this in the codebase somewhere?

```rust
#[cfg(not(all(feature = "occlum_feature", target_arch = "x86_64")))]
compile_error!("occlum only available on Intel x86_64");
```

Reply:
Good comment. I think the best place is `src/snapshots/occlum/mod.rs`. What do you think of it?

Reply:
I think what we actually want is the following (but please test it ;)...

```rust
#[cfg(all(feature = "occlum_feature", not(target_arch = "x86_64")))]
compile_error!("occlum only available on Intel x86_64");
```

Reply:
Ok, get it.